plugins/module_utils/foundation/node_discovery.py

```diff
@@ -43,7 +43,7 @@ def discover(
     def _add_network_details(self, blocks, timeout):
         ipv6_addresses = self._get_ipv6_address(blocks)
         node_network_details = NodeNetworkDetails(self.module)
-        nodes = node_network_details.retrieve(ipv6_addresses, timeout)
+        nodes, _ = node_network_details.retrieve(ipv6_addresses, timeout)
```

Comment:
The interface is changed in master. The entity class fetch_url returns only resp now. Please rebase with master and use the new interface.